ledger-tool/src/bigtable.rs

```diff
@@ -49,19 +49,27 @@ async fn upload(
         force_reupload,
         ..ConfirmedBlockUploadConfig::default()
     };
+    let blockstore = Arc::new(blockstore);
+    let mut last_slot_uploaded = 0;
 
-    solana_ledger::bigtable_upload::upload_confirmed_blocks(
-        Arc::new(blockstore),
-        bigtable,
-        starting_slot,
-        ending_slot,
-        config,
-        Arc::new(AtomicBool::new(false)),
-    )
-    .await
-    .map(|last_slot_uploaded| {
+    loop {
+        if starting_slot > 0 && starting_slot == last_slot_uploaded {
```

Comment:
```suggestion
        if starting_slot >= 0 && starting_slot == last_slot_uploaded {
```
Is there a hard requirement that `starting_slot` not be 0?

Reply:
0 is a valid `starting_slot`; that's why we don't want to break out of the loop there.

Reply:
Oh duh. 😅